.pre-commit-config.yaml

```diff
@@ -81,12 +81,15 @@ repos:
         exclude_types: [file, symlink]
         args:
           [
+            "-M", # Run sphinx in make mode, so we can use -D flag later
+                  # Note: -M requires next 3 args to be builder, source, output
+            "html", # Specify builder
+            "./docs", # Source directory of documents
+            "./_readthedocs", # Output directory for rendered documents
             "-T", # Show full trace back on exception
-            "-E", # Don't use saved env. always read all files.
-            "-b", # Flag to select which builder to use
-            "html", # Use the HTML builder
+            "-E", # Don't use saved env; always read all files
             "-d", # Flag for cached environment and doctrees
-            "docs/build/doctrees", # directory
-            "./docs", # Source directory of documents
-            "docs/build/html", # Output directory for rendered documents.
-          ]
\ No newline at end of file
+            "./docs/_build/doctrees", # Directory
+            "-D", # Flag to override settings in conf.py
+            "exclude_patterns=notebooks/*", # Exclude our notebooks from pre-commit
```

Comment:
The changes here, should speed up the pre-commit checks that run the sphinx build. The primary change here is that jupyter notebooks will no longer be built with each commit. If you all were specifically wanting to check that the jupyter notebooks build for each commit, we should change this back. Also, we now specify the `./docs/_build/` directory for doctrees and `./_readthedocs` for html output. Without that change, we potentially end up in  one of those highly nested directory structure situations where Sphinx eventually fails.